fast64_internal/sm64/sm64_level_writer.py

```diff
@@ -217,13 +217,13 @@ def getOrMakeMacroByLevelName(self, levelName):
 		self.defineMacros.append(macroCmd)
 		return macroCmd
 
-FAST64KEEP = 'FAST64KEEP'
-FAST64END = 'FAST64END'
-FAST64_KEEPS = {
-	'INCLUDES': 'INCLUDES',
-	'SCRIPTS': 'SCRIPTS',
-	'LEVEL_COMMANDS': 'LEVEL_COMMANDS',
-	'AREA_COMMANDS': 'AREA_COMMANDS',
+beginPersistentBlockMagic = 'Fast64 begin persistent block'
+endPersistentBlockMagic = 'Fast64 end persistent block'
+persistentBlocksNames = {
+	'[includes]': '[includes]',
+	'[scripts]': '[scripts]',
+	'[level commands]': '[level commands]',
+	'[area commands]': '[area commands]',
```

Comment:
I don't understand why this needs to have key==value, if it really does then there's no point to this dictionary, right?

If it doesn't need key==value having less magic keys would be better
(or maybe don't use a dict for this, eg `includesPersistentBlockName = "[includes]"`)

Reply:
yeah i'm fine with that, i'm mostly just doing it this way because i wanted to group the blocks together under the same structure, i really actually dont need the keys to equal the values. what do you think about a class for these constants? that way these constants can be accessed from `PersistentBlocks.includes`

Reply:
```python
class PersistentBlocks:
	includes = '[includes]'
	scripts = '[scripts]'
	levelCommands = '[level commands]'
	areaCommands = '[area commands]'

	@classmethod
	def new(cls) -> dict:
		return {
			[cls.includes]: [],
			[cls.scripts]: [],
			[cls.levelCommands]: [],
			[cls.areaCommands]: defaultdict(list)
		}
```

what about something like this? then the LevelScript class can do `self.persistentBlocks = PersistentBlocks.new()`

Reply:
i could even keep the magic strings or regexes in here too, so that the data is all together under a class. but maybe thats overkill lol

Reply:
doesn't look very python-y (more java-y) but I like it

Reply:
yeah it definitely isn't the most python-y implementation, but having loose constants can get hard to manage so i do like to get things like this grouped together

Reply:
okay, updated again!